cbmc_viewer/metric_reportt.py

```diff
@@ -0,0 +1,392 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""CBMC runtime analysis metric report.
+
+This module collects all metric data across multiple proofs and
+generates a single consolidated metric summary report.
+"""
+
+import os
+import re
+import json
+
+import voluptuous
+import voluptuous.humanize
+
+from cbmc_viewer import filet
+from cbmc_viewer import parse
+from cbmc_viewer import templates
+from cbmc_viewer import util
+
+################################################################
+
+VALID_BYTEOP_STATS = voluptuous.schema_builder.Schema({
+    'numOfExtracts': int,
+    'numOfUpdates': int,
+    'link': str
+    },required=True)
+
+VALID_ALIAS_STATS = voluptuous.schema_builder.Schema({
+    'max': int,
+    'total': int,
+    'link': str
+    },required=True)
+
+VALID_MEMOP_STATS = voluptuous.schema_builder.Schema({
+    'count': int,
+    'link': str
+    },required=True)
+
+VALID_ARRAY_CONSTRAINT_STATS = voluptuous.schema_builder.Schema({
+    'count': int,
+    'link': str
+    },required=True)
+
+VALID_CBMC_PROOF_STATS = voluptuous.schema_builder.Schema({
+    'programSteps': int,
+    'vccCount': int,
+    'varCount': int,
+    'clauseCount': int
+    },required=True)
+
+VALID_CLAUSE_STATS = voluptuous.schema_builder.Schema({
+    'clauseCoreCount': int,
+    'link': str
+    },required=True)
+
+VALID_RUNTIME_OBJ = voluptuous.schema_builder.Schema({
+    'val': float,
+    'string': str
+    },required=True)
+
+VALID_RUNTIME = voluptuous.schema_builder.Schema({
+    'symex': VALID_RUNTIME_OBJ,
+    'convertSSA': VALID_RUNTIME_OBJ,
+    'postProc': VALID_RUNTIME_OBJ,
+    'solver': VALID_RUNTIME_OBJ,
+    'maxKey': {
+        'val': str,
+        'string': str
+        }
+    },required=True)
+
+VALID_RUNTIME_ANALYSIS_SUMMARY = voluptuous.schema_builder.Schema([{
+    'jobname': str,
+    'link': str,
+    'byteop': VALID_BYTEOP_STATS,
+    'alias': VALID_ALIAS_STATS,
+    'memop': VALID_MEMOP_STATS,
+    'array': VALID_ARRAY_CONSTRAINT_STATS,
+    'cbmc': VALID_CBMC_PROOF_STATS,
+    'clause': VALID_CLAUSE_STATS,
+    'runtimeKissat': float,
+    'runtime': VALID_RUNTIME
+    }],required=True)
+
+VALID_SUMMARY = voluptuous.schema_builder.Schema({
+    'summary': VALID_RUNTIME_ANALYSIS_SUMMARY
+    },required=True)
+
+################################################################
+
+class RuntimeAnalysisSummary:
+
+    def __init__(self, proofs, srcdir):
+        self.summary = []
+
+        for proof in proofs:
+            jobname = proof
+            proof_path = os.path.abspath(proof)
+            outdir = os.path.join(proof_path, 'html', 'html')
```

Comment:
I think you want the first html to be args.report_dir, because the report directory containing the json and html directories is not guaranteed to be names html.